### PR TITLE
feat: add Video Stitcher samples and tests

### PR DESCRIPTION
**Pull request by irataxy**
b:202972981

Part 2 of 4: CDN keys (see https://cloud.google.com/video-stitcher/docs/how-to/managing-cdn-keys)

### CODE REVIEW

File: media/videostitcher/src/create_cdn_key.php [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+ * @param string  $callingProjectId   The project ID to run the API call under
+ * @param string  $location           The location of the CDN key
+ * @param string  $cdnKeyId           The ID of the CDN key to be created
+ * @param string  $hostname           The hostname of the CDN key
+ * @param string  $keyName            The public name of the CDN key
+ * @param string  $privateKey         For a Media CDN key, this is a 64-byte Ed25519 private
+ *  key encoded as a base64-encoded string. For a Cloud CDN key, this is a base64-encoded
+ *  string secret.
+ * @param boolean $isMediaCdn         If true, create a Media CDN key. If false,
+ *  create a Cloud CDN key.
```
Comment (bshaffer) [inline media/videostitcher/src/create_cdn_key.php:45]: nit:
```suggestion
 * @param string  $privateKey         For a Media CDN key, this is a 64-byte Ed25519 private
 *                                    key encoded as a base64-encoded string. For a Cloud CDN key,
 *                                    this is a base64-encoded string secret.
 * @param boolean $isMediaCdn         If true, create a Media CDN key. If false,
 *                                    create a Cloud CDN key.
```

File: media/videostitcher/src/create_cdn_key_akamai.php [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+/**
+ * Creates an Akamai CDN key.
+ *
+ * @param string  $callingProjectId   The project ID to run the API call under
+ * @param string  $location           The location of the CDN key
+ * @param string  $cdnKeyId           The ID of the CDN key to be created
+ * @param string  $hostname           The hostname of the CDN key
+ * @param string  $tokenKey           The base64-encoded string token key for
+ * the Akamai CDN edge configuration
```
Comment (bshaffer) [inline media/videostitcher/src/create_cdn_key_akamai.php:40]: nit
```suggestion
 * @param string  $tokenKey           The base64-encoded string token key for
 *                                    the Akamai CDN edge configuration
```

File: media/videostitcher/src/update_cdn_key.php [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+ * @param string  $callingProjectId   The project ID to run the API call under
+ * @param string  $location           The location of the CDN key
+ * @param string  $cdnKeyId           The ID of the CDN key to be created
+ * @param string  $hostname           The hostname of the CDN key
+ * @param string  $keyName            The public name of the CDN key
+ * @param string  $privateKey         For a Media CDN key, this is a 64-byte Ed25519 private
+ *  key encoded as a base64-encoded string. For a Cloud CDN key, this is a base64-encoded
+ *  string secret.
+ * @param boolean $isMediaCdn         If true, update a Media CDN key. If false,
+ *  update a Cloud CDN key.
```
Comment (bshaffer) [inline media/videostitcher/src/update_cdn_key.php:46]:
```suggestion
 * @param string  $privateKey         For a Media CDN key, this is a 64-byte Ed25519 private
 *                                    key encoded as a base64-encoded string. For a Cloud CDN key,
 *                                    this is a base64-encoded string secret.
 * @param boolean $isMediaCdn         If true, update a Media CDN key. If false,
 *                                    update a Cloud CDN key.
```

File: media/videostitcher/src/update_cdn_key_akamai.php [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+/**
+ * Updates an Akamai CDN key.
+ *
+ * @param string  $callingProjectId   The project ID to run the API call under
+ * @param string  $location           The location of the CDN key
+ * @param string  $cdnKeyId           The ID of the CDN key to be created
+ * @param string  $hostname           The hostname of the CDN key
+ * @param string  $tokenKey           The base64-encoded string token key for
+ * the Akamai CDN edge configuration
```
Comment (bshaffer) [inline media/videostitcher/src/update_cdn_key_akamai.php:41]:
```suggestion
 * @param string  $tokenKey           The base64-encoded string token key for
 *                                    the Akamai CDN edge configuration
```

File: media/videostitcher/test/videoStitcherTest.php [hunk 5]
```diff
@@ -97,6 +113,126 @@ public function testSlates()
...
+            self::$location,
+            $cdnKeyId
+        ]);
+        $this->assertStringContainsString($cdnKeyName, $output);
+
+        $output = $this->runFunctionSnippet('list_cdn_keys', [
+            self::$projectId,
+            self::$location
+        ]);
+        $this->assertStringContainsString($cdnKeyName, $output);
```
Comment (bshaffer) [inline media/videostitcher/test/videoStitcherTest.php:147]: These tests would be _much better_ if written individually and depending on each other (for example, see the "uptime check" tests in [monitoringTest.php](https://github.com/GoogleCloudPlatform/php-docs-samples/blob/main/monitoring/test/monitoringTest.php#L62-L107)

File: media/videostitcher/test/videoStitcherTest.php [hunk 6]
```diff
@@ -42,12 +42,28 @@ class videoStitcherTest extends TestCase
...
+    private static $updatedCloudCdnKeyName = 'updated-cloud-cdn-key';
+    private static $mediaCdnKeyName = 'media-cdn-key';
+    private static $updatedMediaCdnKeyName = 'updated-media-cdn-key';
+
+    private static $cloudCdnPrivateKey = 'VGhpcyBpcyBhIHRlc3Qgc3RyaW5nLg==';
+    private static $updatedCloudCdnPrivateKey = 'VGhpcyBpcyBhbiB1cGRhdGVkIHRlc3Qgc3RyaW5nLg==';
+    private static $mediaCdnPrivateKey = 'MTIzNDU2Nzg5MDEyMzQ1Njc4OTAxzg5MDEyMzQ1Njc4OTAxMjM0NTY3DkwMTIzNA';
+    private static $updatedMediaCdnPrivateKey = 'ZZZzNDU2Nzg5MDEyMzQ1Njc4OTAxzg5MDEyMzQ1Njc4OTAxMjM0NTY3DkwMTIZZZ';
+    private static $akamaiKey = 'VGhpcyBpcyBhIHRlc3Qgc3RyaW5nLg==';
+    private static $updatedAkamaiKey = 'VGhpcyBpcyBhbiB1cGRhdGVkIHRlc3Qgc3RyaW5nLg==';
```
Comment (bshaffer) [inline media/videostitcher/test/videoStitcherTest.php:58]: I assume these are test keys and don't actually contain any sensitive information, yes?

**Review by bshaffer — CHANGES_REQUESTED** [review]
Everything looks good other than the whitespace. I've also added a request to refactor the tests. Please let me know if you have any questions and I can provide more guidance!

File: media/videostitcher/test/videoStitcherTest.php (hunk 6 above)
Comment (irataxy) [reply media/videostitcher/test/videoStitcherTest.php:58]: Correct, just gibberish or encoded "This is a test string"

File: media/videostitcher/test/videoStitcherTest.php (hunk 5 above)
Comment (irataxy) [reply media/videostitcher/test/videoStitcherTest.php]: Done

**Review by bshaffer — APPROVED** [review]

**Pull request merged**